bin/cron/build_contact_rollups

```diff
@@ -0,0 +1,432 @@
+    ON DUPLICATE KEY UPDATE contact_rollups.name = VALUES(name)"
+  log_completion(start)
+end
+
+def insert_from_pegasus_forms
+  start = Time.now
+  log "Inserting contacts and IP geo data from pegasus.forms"
+  PEGASUS_DB.run "
+    INSERT IGNORE INTO contact_rollups (email, name, city, state, postal_code, country)
+    SELECT email, name, city, state, postal_code, country FROM forms
+    LEFT OUTER JOIN form_geos on form_geos.form_id = forms.id
+    WHERE form_geos.indexed_at IS NOT NULL"
+  log_completion(start)
+end
+
+def update_roles
+  start = Time.now
+  log "Updating Facilitator, Workshop Organizer, District Contact roles from dashboard.users"
+  append_to_role_list_from_permission("facilitator", "Facilitator")
+  append_to_role_list_from_permission("workshop_organizer", "Workshop Organizer")
+  append_to_role_list_from_permission("district_contact", "District Contact")
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+
+  start = Time.now
+  log "Updating Professional Learning Partner role"
+  append_plp_to_role_list
+  log_completion(start)
+end
+
+def append_to_role_list_from_permission(permission_name, dest_value)
+  PEGASUS_DB.run "
+    UPDATE contact_rollups
+    INNER JOIN #{DASHBOARD_DB_NAME}.users AS users ON users.id = contact_rollups.dashboard_user_id
+    INNER JOIN #{DASHBOARD_DB_NAME}.user_permissions AS user_permissions ON user_permissions.user_id = users.id
+    SET roles = CONCAT(COALESCE(CONCAT(roles, ','), ''), '#{dest_value}')
+    WHERE LENGTH(users.email) > 0
+    AND user_permissions.permission = '#{permission_name}' AND contact_rollups.id > 0"
+end
+
+def append_to_list_field_from_form(form_kinds, dest_field, dest_value)
+  start = Time.now
+  log "Appending '#{dest_field}' field with #{dest_value} from forms of kind #{form_kinds}"
+
+  PEGASUS_DB.run "
+    UPDATE contact_rollups
+    INNER JOIN forms ON forms.email = contact_rollups.email
+    SET #{dest_field} = CONCAT(COALESCE(CONCAT(#{dest_field}, ','), ''), #{dest_value})
+    WHERE forms.kind IN (#{form_kinds}) AND contact_rollups.id > 0"
+  log_completion(start)
+end
+
+def append_plp_to_role_list
+  PEGASUS_DB.run "
+    UPDATE contact_rollups
+    INNER JOIN #{DASHBOARD_DB_NAME}.users AS users ON users.id = contact_rollups.dashboard_user_id
+    INNER JOIN #{DASHBOARD_DB_NAME}.professional_learning_partners AS professional_learning_partners ON professional_learning_partners.contact_id = users.id
+    SET roles = CONCAT(COALESCE(CONCAT(roles, ','), ''), 'PLP')
+    WHERE LENGTH(users.email) > 0 AND contact_rollups.id > 0"
+end
+
+def update_courses_facilitated
+  start = Time.now
+  log "Updating courses_facilitated"
+  PEGASUS_DB.run "
+    UPDATE contact_rollups,
+    (SELECT facilitator_id, GROUP_CONCAT(course) AS courses FROM
+      (SELECT DISTINCT facilitator_id, course FROM #{DASHBOARD_DB_NAME}.pd_course_facilitators
+        WHERE COURSE IN (#{COURSE_LIST})
+        ORDER by 1,2
```

Comment:
For general sanity I wanted the comma separated list of course names (built by GROUP_CONCAT) to be alphabetized. It does not strictly need to be, but I prefer it. So I could have done ORDER BY 2. Doing the explicit ordering by both columns helps in debugging (I get consistent results when running the query), which is probably how it got in there in the first place. Given that I may debug this in isolation in the future I will leave the ORDER BY 1,2 in place.